arm-monitor/2017-04-01/swagger/activityLogAlerts_API.json

```diff
@@ -0,0 +1,451 @@
+                "x-ms-examples": {
+                    "List activity log alerts": {
+                        "$ref": "../examples/listActivityLogAlerts.json"
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": null
+                },
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The request succeeded.",
+                        "schema": {
+                            "$ref": "#/definitions/ActivityLogAlertList"
+                        }
+                    },
+                    "default": {
+                        "description": "An error occurred and the list of activity log alerts could not be retrieved.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/microsoft.insights/activityLogAlerts": {
+            "get": {
+                "description": "Get a list of all activity log alerts in a resource group.",
+                "operationId": "ActivityLogAlerts_ListByResourceGroup",
+                "x-ms-examples": {
+                    "List activity log alerts": {
+                        "$ref": "../examples/listActivityLogAlerts.json"
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": null
+                },
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "The request succeeded.",
+                        "schema": {
+                            "$ref": "#/definitions/ActivityLogAlertList"
+                        }
+                    },
+                    "default": {
+                        "description": "An error occurred and the list of activity log alerts could not be retrieved.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "definitions": {
+        "Resource": {
```

Comment:
A. Yes, I won't block you on this one.
B. Yes, consolidating definitions in a common document sounds ideal. I'm not sure what you mean by root folder, as long as you can specify a relative path you should be fine. Having said that you might wish to hold off a little given some upcoming directory structure changes here.
C. The document can be just like any other swagger document but devoid of paths section but could contain almost of the common definitions.